src/GCPStorageService.js

```diff
@@ -299,14 +299,20 @@ export class GCPStorageService extends BaseStorageService {
    * @description                     - Generates a downloadable URL for a file in the gcloud bucket.
    * @param {string} container        - gcloud bucket name.
    * @param {string} filePath         - Path to the file in the bucket.
+   * @param {number} expiresIn        - Expiry time for the signed URL in seconds. Default is 3600.
+   * @param {string} permission       - Permission for the operation. Use WRITE for PUT operations.
    * @returns {Promise<string>}       - A downloadable URL for the specified file.
    */
-  async getDownloadableUrl(container, filePath) {
+  async getDownloadableUrl(container, filePath,expiresIn = 3600, permission = '') {
```

Comment:
instead of keeping empty "" makeit by default read permission for getdownloadable url

Reply:
@praveenKDass

Reply:
No, For public bucket we don't need to call preSignedURl, so it will be empty string if added "r" as a default , it will be calling preSignedURl  .We don't need that so we will keep it as empty string @ankit1shahu